core/src/main/java/org/mule/context/notification/DefaultFlowCallStack.java

```diff
@@ -31,23 +31,29 @@ public DefaultFlowCallStack()
 
     public DefaultFlowCallStack(FlowCallStack flowCallStack)
     {
-        if(flowCallStack != null)
+        if (flowCallStack != null && flowCallStack instanceof DefaultFlowCallStack)
```

Comment:
If we are adding an interface for FlowCallStack and we need to do this cast, then something is wrong